src/acl_usm.cpp

```diff
@@ -606,6 +612,14 @@ CL_API_ENTRY cl_int CL_API_CALL clGetMemAllocInfoINTEL(
     }
   } break;
 
+  case CL_MEM_ALLOC_BUFFER_LOCATION_INTEL: {
+    if (usm_alloc) {
+      RESULT_UINT(usm_alloc->mem->mem_id);
+    } else {
+      RESULT_UINT(0);
```

Comment:
Is `0` the "ID corresponding to default global memory"? If so, this would be a suitable comment here.

Reply:
Technically there isn't an buffer_location info of a null or invalid usm. But for any valid usm that do not have property specified, it is gotten from `acl_get_default_memory(device->def)` Which is the memory in board_spec.xml annotated with default.

So default memory is not necessarily 0, its just that when usm is null, there is no information about it, so I just return 0.

Reply:
Thanks @sherry-yuan. Could you clarify with the `buffer_location` specification contributors what the desired behaviour would be when the pointer does not point to a valid usm allocation? Should an error be returned instead? Does the specification need clarification?

Reply:
I think it is fine to return 0, because global mem 0 always exist in the board spec, and for user who expect non zero answer, they probably already know something is wrong with their pointer when it is not. Further it is almost a convention that developers checks for validity of the usm pointers before running any other function on it.

I don't think there need to be a spec change, as 0 is the assumed default mem (when no other mem are annotated).

Reply:
Another reason is this spec is general to all vendors. Other vendor's default mem may have other values, which then it doesn't make sense to forcefully set it 0 in this case.